scene/resources/particles_material.cpp

```diff
@@ -484,6 +487,12 @@ void ParticlesMaterial::_update_shader() {
 		code += "	float color_initial_rand = rand_from_seed(alt_seed);\n";
 	}
 
+	if (emission_shape == EMISSION_SHAPE_POINTS || emission_shape == EMISSION_SHAPE_DIRECTED_POINTS) {
+		code += "	int point = min(emission_texture_point_count - 1, int(rand_from_seed(alt_seed) * float(emission_texture_point_count)));\n";
+		code += "	ivec2 emission_tex_size = textureSize(emission_texture_points, 0);\n";
+		code += "	emission_tex_ofs = ivec2(point % emission_tex_size.x, point / emission_tex_size.x);\n";
+	}
+
```

Comment:
This should be under:
```
	code += "	float pi = 3.14159;\n";
	code += "	float degree_to_rad = pi / 180.0;\n";
	code += "\n";
```
So that it is consistent with the ``start()`` function. 

I.e. it should look like this:
```
        code += "	float pi = 3.14159;\n";
	code += "	float degree_to_rad = pi / 180.0;\n";
	code += "\n";

	if (emission_shape == EMISSION_SHAPE_POINTS || emission_shape == EMISSION_SHAPE_DIRECTED_POINTS) {
		code += "	int point = min(emission_texture_point_count - 1, int(rand_from_seed(alt_seed) * float(emission_texture_point_count)));\n";
		code += "	ivec2 emission_tex_size = textureSize(emission_texture_points, 0);\n";
		code += "	ivec2 emission_tex_ofs = ivec2(point % emission_tex_size.x, point / emission_tex_size.x);\n";
		code += "	ivec2 emission_tex_ofs = ivec2(point % emission_tex_size.x, point / emission_tex_size.x);\n";
	}
```

Reply:
Done.

Reply:
Looks like you forgot to remove it, and now you have two copies of (nearly) the same code.

Reply:
I removed it.